app/javascript/packages/document-capture/hooks/use-cookie.js

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+
+function getCookieObject() {
+  const { cookie } = document;
+  return cookie
+    .split(';')
+    .map((part) => part.trim().split('='))
+    .filter(([_key, value]) => typeof value === 'string')
+    .reduce((result, [key, value]) => Object.assign(result, { [key]: value }), {});
+}
+
+function useCookie(name) {
+  const getCookieValue = () => getCookieObject()[name];
+  const [value, setStateValue] = useState(getCookieValue);
+
+  /**
+   * @param {string?} nextValue Value to set, or null to delete the value.
+   */
+  function setValue(nextValue) {
+    const cookieValue = nextValue === null ? '; Max-Age=0' : nextValue;
+    document.cookie = `${name}=${cookieValue}`;
+    setStateValue(nextValue);
+  }
+
+  useEffect(() => {
+    const originalCookieDescriptor = Object.getOwnPropertyDescriptor(Document.prototype, 'cookie');
+    Object.defineProperty(Document.prototype, 'cookie', {
+      ...originalCookieDescriptor,
+      set(nextValue) {
+        originalCookieDescriptor?.set?.call(this, nextValue);
+        setStateValue(getCookieValue);
+      },
+    });
+
+    return () => {
+      Object.defineProperty(
+        Document.prototype,
+        'cookie',
+        /** @type {PropertyDescriptor} */ (originalCookieDescriptor),
+      );
```

Comment:
I have not seen this pattern before, is this essentially monkeypatching `document.cookie`, and is this just to work around the acuant SDK writing to the cookie?

Reply:
> I have not seen this pattern before, is this essentially monkeypatching `document.cookie`, and is this just to work around the acuant SDK writing to the cookie?

Yeah, this is a workaround to the fact that there are no events for us to know when a cookie has changed. For our [bypass logic](https://github.com/18F/identity-idp/blob/aduth-lg-4392-acuant-11-5/app/javascript/packages/document-capture/components/acuant-capture.jsx) to work, it needs to have access to the latest value of the cookie, since Acuant may have just assigned it.

An alternative is that we can check the value of the cookie right inline right there, i.e. `document.cookie.includes('AcuantCameraHasFailed=')`, which is definitely more straight-forward, but leaves us in an awkward spot whether we want to have this hook managing those cookie interactions.

Future support for [CookieStore](https://developer.mozilla.org/en-US/docs/Web/API/CookieStore) would make this a lot easier. Maybe there's a polyfill we could use, to at least avoid us from rolling our own? 🤔

Reply:
> Maybe there's a polyfill we could use, to at least avoid us from rolling our own change handling? 🤔

From the ones I can find, they are limited in only supporting change events from changes made via the API (`CookieStore#set`), which we can't rely on for Acuant-assigned cookie values.